File: src/main/java/leets/weeth/domain/penalty/application/dto/PenaltyDTO.java
package leets.weeth.domain.penalty.application.dto;

import jakarta.validation.constraints.NotNull;
import leets.weeth.domain.penalty.domain.entity.enums.PenaltyType;
import lombok.Builder;

import java.time.LocalDateTime;
import java.util.List;

public class PenaltyDTO {

    @Builder
    public record Save(
            @NotNull Long userId,
            @NotNull PenaltyType penaltyType,
            String penaltyDescription
    ){}

    @Builder
    public record Update(
            @NotNull Long penaltyId,
            String penaltyDescription
    ){}

    @Builder
    public record ResponseAll(
            Integer cardinal,
            List<Response> responses
    ){}

    @Builder
    public record Response(
        Long userId,
        Integer penaltyCount,
        Integer warningCount,
        String name,
        List<Integer> cardinals,
        List<PenaltyDTO.Penalties> Penalties
    ){}

    @Builder
    public record Penalties(
       Long penaltyId,
       PenaltyType penaltyType,
       Integer cardinal,
       String penaltyDescription,
       LocalDateTime time
    ){}

}

File: src/main/kotlin/com/weeth/domain/penalty/application/dto/PenaltyDTO.kt
package com.weeth.domain.penalty.application.dto

import com.weeth.domain.penalty.domain.entity.enums.PenaltyType
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime

class PenaltyDTO {
    data class Save(
        @field:NotNull
        val userId: Long,
        @field:NotNull
        val penaltyType: PenaltyType,
        val penaltyDescription: String?,
    )

    data class Update(
        @field:NotNull
        val penaltyId: Long,
        val penaltyDescription: String?,
    )

    data class ResponseAll(
        val cardinal: Int?,
        val responses: List<Response>,
    )

    data class Response(
        val userId: Long?,
        val penaltyCount: Int?,
        val warningCount: Int?,
        val name: String?,
        val cardinals: List<Int>,
        val penalties: List<Penalties>,
    )

    data class Penalties(
        val penaltyId: Long?,
        val penaltyType: PenaltyType?,
        val cardinal: Int?,
        val penaltyDescription: String?,
        val time: LocalDateTime?,
    )
}
